modules/nextflow/src/main/groovy/nextflow/trace/WebLogObserver.groovy

```diff
@@ -188,40 +198,63 @@ class WebLogObserver implements TraceObserver{
         message.runStatus = event // deprecated to be removed
         message.utcTime = time
 
-        // Append the trace object if present
-        if (trace)
-            message.trace = trace.store
+        if (payload instanceof TraceRecord)
+                message.trace = (payload as TraceRecord).store
+        else if (payload instanceof Map)
+                message.metadata = payload as Map
+        else if (payload != null)
+            throw new IllegalArgumentException("Only TraceRecord and Manifest class types are supported: [${payload.getClass().getName()}] $payload")
+
+        /*
+        Make sure that:
+            1) Path objects are converted to String (or it throws a StackOverflow error)
+            2) Dates follow ISO 8601
+            3) Durations are in millisecond
+         */
+        def generator = new JsonGenerator.Options()
+                .addConverter(Path) { Path p, String key ->
+                        p.toString()
+                    }
+                .addConverter(Duration) { Duration d, String key ->
+                        d.durationInMillis
+                    }
+                .dateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
+                .timezone("UTC")
+                .build()
 
```

Comment:
The `generator` should be created in the constructor, otherwise a new instance is created for each message.